src/App.tsx

```diff
@@ -31,65 +31,135 @@ export function getReorderedGoods(
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+  isSortedAlphabetically: boolean,
+  isSortedByLength: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+    isSortedAlphabetically: false,
+    isSortedByLength: false,
+  };
+
+  sortByAlphabet = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+      isSortedAlphabetically: true,
+      isSortedByLength: false,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+      isSortedAlphabetically: false,
+      isSortedByLength: true,
+    });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+      isSortedAlphabetically: false,
+      isSortedByLength: false,
+    });
+  };
+
+  render() {
+    const {
+      sortType,
+      isReversed,
+      isSortedAlphabetically,
+      isSortedByLength,
+    } = this.state;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info ${!isSortedAlphabetically ? 'is-light' : ''}`}
+            onClick={this.sortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={`button is-info ${!isSortedByLength ? 'is-light' : ''}`}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={`button is-info ${!isReversed ? 'is-light' : ''}`}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          { (isReversed || isSortedAlphabetically || isSortedByLength) && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={this.reset}
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {getReorderedGoods(goodsFromServer, { sortType, isReversed })
```

Comment:
try to avoid calculations inside the render, let's move the result to a variable instead